src/AutoHashEqualsCached.jl

```diff
@@ -69,135 +88,209 @@ function unpack_name(node::Expr)
     end
 end
 
-unpack_type_name(n::Symbol) = (n, n, nothing)
-function unpack_type_name(n::Expr)
+unpack_type_name(__source__, n::Symbol) = (n, n, nothing)
+function unpack_type_name(__source__, n::Expr)
     if n.head === :curly
```

Comment:
General heads up for a possible future PR (definitely not this one 😅), but using `@capture` from the `MacroTools` package might make these a lot cleaner. I haven't used it much myself, but it made another macro on the RAICode side a loooot more readable - see https://github.com/RelationalAI/raicode/pull/14088

Reply:
I avoided it because, among other things, it skips the line number nodes, making it harder to report good diagnostics.